packages/parser-javascript/src/parser.ts

```diff
@@ -1,229 +1,70 @@
-             *
-             * Continuing with the previous example, this code will execute:
-             *
-             * acornWalk.simple(ast, {
-             *     CallExpresion(node) {
-             *         // code from CallExpression in hint 1
-             *         // code from CallExpression in hint 2
-             *     },
-             *     Literal(node) {
-             *         // code from Literal in hint 3
-             *     }
-             * });
-             *
-             * acornWalk.full(ast, (node) => {
-             *     // code from callback in hint 4
-             * });
-             */
-            Object.entries(walkArrays).forEach(([methodName, walkArray]) => {
-                walkArray.forEach(([{ node, state, base }, visitors]) => {
-                    let allVisitors: NodeVisitor | Function = {};
-
-                    if (visitors.has(defaultCallbacksProperty)) {
-                        // `full` and `fullAncestor` only track an array of callbacks.
-                        const callbacks = visitors.get(defaultCallbacksProperty)!;
-
-                        /* istanbul ignore next */
-                        allVisitors = (callbackNode: ESTree.Node, callbackState: any, typeOrAncestors: string | ESTree.Node[]) => {
-                            callbacks.forEach((callback: Function) => {
-                                callback(callbackNode, callbackState, typeOrAncestors);
-                            });
-                        };
-                    } else {
-                        // `ancestor` and `simple` track an array of NodeVisitors which need merged.
-                        for (const [name, callbacks] of visitors) {
-                            /* istanbul ignore next */
-                            (allVisitors as any)[name] = (callbackNode: ESTree.Expression, ancestors?: ESTree.Node[]) => {
-                                callbacks.forEach((callback: Function) => {
-                                    callback(callbackNode, ancestors);
-                                });
-                            };
-                        }
-                    }
-
-                    acornWalk[methodName](node, allVisitors, base, state);
+            const options: Options = { locations: true, ranges: true };
+            const ast = parser.parse(sourceCode, options) as Node;
+            const tokens = [...parser.tokenizer(sourceCode, options)];
+
+            await combineWalk(async (walk) => {
+                await this.engine.emitAsync(`parse::end::javascript`, {
+                    ast,
+                    element,
+                    resource,
+                    sourceCode,
+                    tokens,
+                    walk
                 });
             });
+
         } catch (err) {
-            logger.error(`Error parsing JS code: ${sourceCode}`);
+            logger.error(`Error parsing JS code (${err}): ${sourceCode}`);
```

Comment:
Do we want to do this via `logger` or `debug`? It's not like the execution is being interrupted so I'm more inclined to use `debug`.

Reply:
I can change this to `debug`, though I'll admit I'm not clear on when we prefer using one over the other.

Reply:
The undocumented feedback in my head is:

1. Use `logger` if you want the user to see it 
1. Use `debug` if we (devs of webhint) want to see it when running `--debug`

Unless it's an error that's going to break the execution or that can help the user figure out why something isn't working (generally) I tend to use `debug`.

One thing we could do in the future is to generate a `.log` file automatically (without `--debug`) somewhere. If something fails then we point the user there to look into it or upload it into an issue.